ashpd-backend-demo/src/external_window.rs

```diff
@@ -0,0 +1,51 @@
+use ashpd::WindowIdentifierType;
+use gtk::prelude::*;
+use gtk::{gdk, glib};
+
+use crate::external_wayland_window::ExternalWaylandWindow;
+use crate::external_x11_window::ExternalX11Window;
+
+pub(crate) enum ExternalWindow {
```

Comment:
I believe this enum is useless. All we need is to `impl From<WindowIdentifierType>  for gtk::Window`, no need to split into separate structs, could be just separate functions that do all the calls for you.

Reply:
Once that is done, please move it to `window_identifier` module inside `ashpd` itself and behind `gtk3_wayland`/`gtk3_x11` / `gtk4_wayland` / `gtk4_x11` features.

Reply:
That is a part that we could merge separately from this PR in order to simplify the review process of the backend.

Reply:
I think its more involved, you need to create a window, parent the surface and keep a display.